### ISSUE TITLE
Serverless Docker Registry production setup

### ISSUE THREAD

**Issue by michal-hudy** [kyma-project/kyma#7358]
**Description**

The current docker registry setup works only in a specific setup and has some limitations. Two things definitely need to be addressed:
 - Docker Registry password must be generated with every chart update
 - It must be possible to download images when Kyma is installed with a self-signed certificate
   - On production created solution shouldn't be executed
   - >**HINT**: Use DaemonSet
   - Check on AKS, GKE, Minikube and Gardener

**Reasons**

Migration from Kubeless to KNative and Tekton

**Comment by aerfio** [comment]
Useful info: https://docs.docker.com/engine/security/certificates/#understand-the-configuration, 
https://docs.docker.com/registry/insecure/#use-self-signed-certificates

**Comment by aerfio** [comment]
If you choose the way using daemonset you can use something like this:
```
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: self-signed-cert
  namespace: kyma-system
spec:
  selector:
    matchLabels:
      app: registry-etc-hosts-update
  template:
    metadata:
      labels:
        app: registry-etc-hosts-update
      annotations:
        sidecar.istio.io/inject: "false"
    spec:
      initContainers:
        - name: update
          image: debian:stretch-slim
          volumeMounts:
            - name: etckube
              mountPath: /kube-etc/
              readOnly: false
          command:
            - bash
            - -ce
            - |
              NL=$'\n'

              ls -la /kube-etc/
              echo "Done."
      containers:
        - name: init-container-did-the-work
          image: gcr.io/google_containers/pause-amd64:3.1@sha256:59eec8837a4d942cc19a52b8c09ea75121acc38114a2c68b98983ce9356b8610
      terminationGracePeriodSeconds: 30
      volumes:
        - name: etckube
          hostPath:
            path: /etc/
```

**Comment by aerfio** [comment]
https://github.com/kubernetes/test-infra/blob/master/prow/cluster/tune-sysctls_daemonset.yaml

**Comment by aerfio** [comment]
https://www.ctrl-alt-del.cc/2018/11/solution-rancher-2-k8s-private-registry.html

**Comment by michal-hudy** [comment]
Hm... On GKE it is working, but unfortunately it doesn't work on Minikube (MacOS). I got `Failed to pull image "registry.kyma.local/registry/test-aha:86c1e88f7b08b6e5165fcf4cb72bf21a670aa244acfa00f567b9e0a726bceb18": rpc error: code = Unknown desc = Error response from daemon: Get https://registry.kyma.local/v2/: dial tcp: lookup registry.kyma.local on 192.168.64.1:53: no such host`

I have installed Kyma with scripts and manually added `registry.kyma.local` to hosts (the script is not updated...). Also, I can access the registry in the browser.
